specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/virtualWan.json

```diff
@@ -9713,6 +9929,58 @@
         "name": "VnetLocalRouteOverrideCriteria",
         "modelAsString": true
       }
+    },
+    "SharedKey": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SharedKeyProperties",
+          "description": "Properties of the VpnLink connection shared key."
+        },
+        "etag": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "VpnLink connection shared key Resource."
+    },
+    "SharedKeyProperties": {
```

Comment:
If this is considered a secret, you must not return it in a get API. Instead you need to return it in response to a post action

Reply:
Also remove the ARMChangesRequested label after u respond